Reactive Monolith does not compile

<!--
- Please follow the issue template below for bug reports and feature requests.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

When you create a reactive monolith, it does not compile.

Caught by the [Ionic4J CI process](https://travis-ci.org/github/oktadeveloper/generator-jhipster-ionic/jobs/661960142#L2461).

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile 
(default-compile) on project travis-mongo: Compilation failure
[ERROR] /Users/mraible/reactive-oauth2/src/main/java/io/github/jhipster/
travis/repository/UserRepository.java:[24,5] 
org.springframework.cache.annotation.Cacheable is not a repeatable annotation type
```

##### **Motivation for or Use Case**

Creating a reactive monolith should compile.

##### **Reproduce the error**

Create an app from the master branch using the following JDL. Then run `./mvnw verify`.

```
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "reactive": true,
    "baseName": "travisMongo",
    "packageName": "io.github.jhipster.travis",
    "packageFolder": "io/github/jhipster/travis",
    "authenticationType": "oauth2",
    "enableHibernateCache": false,
    "cacheProvider": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "mongodb",
    "devDatabaseType": "mongodb",
    "prodDatabaseType": "mongodb",
    "searchEngine": false,
    "useSass": true,
    "buildTool": "maven",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "fr"],
    "testFrameworks": ["gatling", "protractor"],
    "serverPort": "8080",
    "jhiPrefix": "jhi",
    "serviceDiscoveryType": false,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "jhipsterVersion": "6.8.0",
    "messageBroker": false,
    "embeddableLaunchScript": false,
    "clientTheme": "none",
    "creationTimestamp": 1584118448298,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "blueprints": []
  }
}
```

##### **Suggest a Fix**

The `UserRepository.java` class that's generated looks as follows:

```java
package io.github.jhipster.travis.repository;

import io.github.jhipster.travis.domain.User;

import org.springframework.cache.annotation.Cacheable;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;


/**
 * Spring Data MongoDB repository for the {@link User} entity.
 */
@Repository
public interface UserRepository extends ReactiveMongoRepository<User, String> {

    String USERS_BY_LOGIN_CACHE = "usersByLogin";

    String USERS_BY_EMAIL_CACHE = "usersByEmail";

    @Cacheable(cacheNames = USERS_BY_EMAIL_CACHE)
    @Cacheable(cacheNames = USERS_BY_LOGIN_CACHE)
    Mono<User> findOneByLogin(String login);

    Flux<User> findAllByLoginNot(Pageable pageable, String login);

    Mono<Long> countAllByLoginNot(String anonymousUser);
}
```

##### **JHipster Version(s)**

`master` branch

--- a/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
@@ -142,12 +142,12 @@ public interface UserRepository extends <% if (databaseType === 'sql') { %>JpaRe
     <%= optionalOrMono %><<%= asEntity('User') %>> findOneByResetKey(String resetKey);
     <%_ } _%>
 
+    <%_ if (authenticationType !== 'oauth2') { _%>
     <%_ if (databaseType === 'couchbase' || databaseType === 'mongodb') { _%>
         <%_ if (cacheManagerIsAvailable === true) { _%>
     @Cacheable(cacheNames = USERS_BY_EMAIL_CACHE)
         <%_ } _%>
     <%_ } _%>
-    <%_ if (authenticationType !== 'oauth2') { _%>
     <%= optionalOrMono %><<%= asEntity('User') %>> findOneByEmailIgnoreCase(String email);
 
     <%_ } _%>

